Mu2eKinKal/src/LoopHelixFit_module.cc

```diff
@@ -318,12 +342,53 @@ namespace mu2e {
       kkbf_ = std::move(std::make_unique<KKBField>(*bfmgr,*det));
     }
     if(print_ > 0) kkbf_->print(std::cout);
+
+    // Print the fit direction configuration
+    if(print_ > 0) printf("[LoopHelixFit::%s::%s] Fit dz/dt direction = %.0f, PDG = %i, use helix slope = %o with threshold %.1f\n",
+                          __func__, moduleDescription().moduleLabel().c_str(), fdir_.dzdt(), (int) fpart_,
+                          useHelixSlope_, (useHelixSlope_) ? slopeSigThreshold_ : -1.f);
   }
 
-  void LoopHelixFit::produce(art::Event& event ) {
-    // empty collections
-    static MEASCOL nohits; // empty
-    static EXINGCOL noexings; // empty
+  std::vector<HelixRecoDir::PropDir> LoopHelixFit::chooseHelixDir(HelixSeed const& hseed) const {
+    if(!goodHelix(hseed)) return {}; //determine if this seed should be fit
+
+    // if using the helix slope to decide the fit direction, check its significance
+    if(useHelixSlope_) {
+      auto predicted_dir = hseed.recoDir().predictDirection(slopeSigThreshold_);
+      if(predicted_dir == HelixRecoDir::PropDir::ambiguous)
+        return {HelixRecoDir::PropDir::downstream, HelixRecoDir::PropDir::upstream};
+      else return {predicted_dir};
+    }
+    // using a forced fit direction
+    if(fdir_.fitDirection() == TrkFitDirection::FitDirection::upstream  ) return {HelixRecoDir::PropDir::upstream  };
+    if(fdir_.fitDirection() == TrkFitDirection::FitDirection::downstream) return {HelixRecoDir::PropDir::downstream};
+    // shouldn't reach this line
+    return {HelixRecoDir::PropDir::downstream, HelixRecoDir::PropDir::upstream};
+  }
+
+  int LoopHelixFit::compare_tracks(const std::unique_ptr<KKTRK>& ktrk_1, const std::unique_ptr<KKTRK>& ktrk_2) const {
+    // return 1 to select the first track, 0 to select the second, -1 to indicate both are bad tracks
```

Comment:
Please code these return values as a (private) enum, so that downstream interpretation will be self-evident;

Reply:
I updated this comparison to use the private enums `kFirstPass`, `kSecondPass`, and `kBothFail` to be more clear.